### PR TITLE
Update information provider and resources

### PR DESCRIPTION
**Pull request by MaxWies**
Information (inclusive Address) pf Provider/Consumable/Device/Personal can be changed now and get updated in the DB.

### CODE REVIEW

File: Pirat.DatabaseTests/ResourceChangeTest.cs [hunk 1]
```diff
@@ -80,68 +80,102 @@ public void Dispose()
...
+
+            //Assert
             Assert.NotNull(response);
             Assert.NotEmpty(response);
             Provider providerFromDevice = response.First().provider;
             Console.Out.WriteLine(providerFromDevice);
             Console.Out.WriteLine(provider);
-            Assert.True(providerFromDevice.Equals(provider));
+            //TODO: 1. Provider is null because so far everything is non public
+            //TODO: 2. If the provider is non-public we should create a provider object with empty attributes instead
```
Comment (chaoran-chen) [inline Pirat.DatabaseTests/ResourceChangeTest.cs:112]: Wait, why that? Please don't start sending some empty provider objects to the frontend. If the provider is non-public, null would be - in my opinion - the correct value.

File: Pirat/Services/Resource/ResourceUpdateService.cs [hunk 2]
```diff
@@ -132,24 +134,128 @@ public Task delete(string token)
...
+            AddressEntity location = new AddressEntity().build(provider.address);
+            _addressMaker.SetCoordinates(location);
+
+            var query = from o in _context.offer
+                join ap in _context.address on o.address_id equals ap.id
+                where o.token == token
+                select new {o, ap};
+
+            var l = query
+                .Select(x => x)
```
Comment (chaoran-chen) [inline Pirat/Services/Resource/ResourceUpdateService.cs:149]: This line is not needed.

File: Pirat/Services/Resource/ResourceUpdateService.cs [hunk 3]
```diff
@@ -132,24 +134,128 @@ public Task delete(string token)
...
+                .ToList();
+            l.ForEach(collection =>
```
Comment (chaoran-chen) [inline Pirat/Services/Resource/ResourceUpdateService.cs:151]: If the token exists, there should be exactly one offer, right? Using ForEach suggests that there could be multiple offers.

File: Pirat/Services/Resource/ResourceUpdateService.cs [hunk 4]
```diff
@@ -132,24 +134,128 @@ public Task delete(string token)
...
+
+            AddressEntity location = new AddressEntity().build(consumable.address);
+            _addressMaker.SetCoordinates(location);
+
+            var query = from o in _context.offer
+                join c in _context.consumable on o.id equals c.offer_id
+                join ac in _context.address on c.address_id equals ac.id
+                where o.token == token
+                select new {o, c, ac};
+
```
Comment (chaoran-chen) [inline Pirat/Services/Resource/ResourceUpdateService.cs:181]: If I am not mistaken, this function is going to change every consumable belonging to an offer? In the where-clause, a `c.id = consumable.id` is missing?

File: Pirat/Services/Resource/ResourceUpdateService.cs (hunk 3 above)
Comment (MaxWies) [reply Pirat/Services/Resource/ResourceUpdateService.cs]: ForEach makes it easier to change the attributes in one expression and query is an enumerable + after this expression checks are made if too many rows are changed (which never should happen).

**Pull request merged**